docs/release-info/release-notes.md

```diff
@@ -57,50 +57,734 @@ For tips about how to write a good release note, see [Release notes](https://git
+* Fixed an issue where concurrent replace skipped intervals locked by append locks during compaction [#16316](https://github.com/apache/druid/pull/16316)
+* Improved error messages when supervisor's checkpoint state is invalid [#16208](https://github.com/apache/druid/pull/16208)
+* Improved serialization of `TaskReportMap` [#16217](https://github.com/apache/druid/pull/16217)
+* Improved compaction segment read and published fields to include sequential compaction tasks [#16171](https://github.com/apache/druid/pull/16171)
+* Improved kill task so that it now accepts an optional list of unused segment versions to delete [#15994](https://github.com/apache/druid/pull/15994)
+* Improved logging when ingestion tasks try to get lookups from the Coordinator at startup [#16287](https://github.com/apache/druid/pull/16287)
+* Improved ingestion performance by parsing an input stream directly instead of converting it to a string and parsing the string as JSON [#15693](https://github.com/apache/druid/pull/15693)
+* Improved the creation of input row filter predicate in various batch tasks [#16196](https://github.com/apache/druid/pull/16196)
+* Improved how Druid fetches tasks from the Overlord to redact credentials [#16182](https://github.com/apache/druid/pull/16182)
+* Optimized `isOvershadowed` when there is a unique minor version for an interval [#15952](https://github.com/apache/druid/pull/15952)
+* Removed `EntryExistsException` thrown when trying to insert a duplicate task in the metadata store&mdash;Druid now throws a `DruidException` with error code `entryAlreadyExists` [#14448](https://github.com/apache/druid/pull/14448)
+* The task status output for a failed task now includes the exception message [#16286](https://github.com/apache/druid/pull/16286)
+
+### SQL-based ingestion
+
+#### Manifest files for MSQ task engine exports
+
+Export queries that use the MSQ task engine now also create a manifest file at the destination, which lists the files created by the query.
+
+During a rolling update, older versions of workers don't return a list of exported files, and older Controllers don't create a manifest file.
+Therefore, export queries ran during this time might have incomplete manifests.
+
+[#15953](https://github.com/apache/druid/pull/15953)
+
+#### `SortMerge` join support
+
+Druid now supports `SortMerge` join for `IS NOT DISTINCT FROM` operations.
+
+[#16003](https://github.com/apache/druid/pull/16003)
+
+#### State of compaction context parameter
+
+Added a new context parameter `storeCompactionState`.
+When set to `true`, Druid records the state of compaction for each segment in the `lastCompactionState` segment field.
+
+[#15965](https://github.com/apache/druid/pull/15965)
+
+#### Selective loading of lookups
+
+Druid now supports selective loading of lookups so that MSQ task engine workers don't load unnecessary lookups.
```

Comment:
This PR does not make the change for MSQ tasks, only kill tasks. The change to MSQ tasks is in a followup PR, which is not in Druid 30.

cc: @Akshat-Jain Please confirm this

Reply:
@adarshsanjeev Yes that's correct.